### PR TITLE
fix(model): #CRRE-550 upgrade model to take joins into account

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Changed the OrderUniversal model to use models instead of ids.
Refactoring of the campaign model to be consistent with other models.

## Checklist tests
Ability to create a campaign.
Ability to modify a campaign.

## Issue ticket number and link
[CRRE-550](https://jira.support-ent.fr/browse/CRRE-550)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/sql/017-crre-550-group-order-table-in-one-view.sql
Comment (BastienBenard) [inline src/main/resources/sql/017-crre-550-group-order-table-in-one-view.sql:1]: Ajout de order_client_id et order_region_id

File: src/main/java/fr/openent/crre/controllers/CampaignController.java [hunk 1]
```diff
@@ -74,10 +77,20 @@ public void campaign(HttpServerRequest request) {
...
+            UserUtils.getUserInfos(eb, request, userInfos -> {
+                try {
+                    campaignService.create(new Campaign(campaign), IModelHelper.toList(campaign.getJsonArray(Field.GROUPS), StructureGroupModel.class))
+                            .onSuccess(campaignResult -> {
+                                Renders.renderJson(request, campaignResult.toJson());
+                                Logging.insert(userInfos, Contexts.CAMPAIGN.toString(), Actions.CREATE.toString(), (String) null, campaign);
+                            })
+                            .onFailure(error -> Renders.renderError(request));
+                } catch (ClassCastException e) {
+                    log.error(" An error occurred when casting campaign id", e);
```
Comment (flmartineau) [inline src/main/java/fr/openent/crre/controllers/CampaignController.java:90]: Ajoute [CampaignController@%s::create] pour être plus clair dans le log.

File: src/main/java/fr/openent/crre/controllers/CampaignController.java [hunk 2]
```diff
@@ -107,17 +120,19 @@ public void updateAccessibility(final HttpServerRequest request) {
...
+                try {
+                    Integer id = Integer.parseInt(request.params().get(Field.ID));
+                    campaignService.update(new Campaign(campaign).setId(id), IModelHelper.toList(campaign.getJsonArray(Field.GROUPS), StructureGroupModel.class))
+                            .onSuccess(campaignResult -> {
+                                Renders.renderJson(request, campaignResult.toJson());
+                                Logging.insert(userInfos, Contexts.CAMPAIGN.toString(), Actions.UPDATE.toString(), id.toString(), campaign);
+                            })
+                            .onFailure(error -> Renders.renderError(request));
+                } catch (ClassCastException e) {
+                    log.error(" An error occurred when casting campaign id", e);
```
Comment (flmartineau) [inline src/main/java/fr/openent/crre/controllers/CampaignController.java:133]: Pareil

**Review by flmartineau — APPROVED** [review]

File: src/main/java/fr/openent/crre/controllers/CampaignController.java (hunk 1 above)
Comment (BastienBenard) [reply src/main/java/fr/openent/crre/controllers/CampaignController.java:90]: J'ai pas mis de réponse en plus

File: src/main/java/fr/openent/crre/controllers/CampaignController.java [hunk 3]
```diff
@@ -74,10 +77,22 @@ public void campaign(HttpServerRequest request) {
     @ResourceFilter(AdministratorRight.class)
     @Override
     public void create(final HttpServerRequest request) {
-        RequestUtils.bodyToJson(request, pathPrefix + "campaign",
-                campaign -> campaignService.create(campaign,
-                        Logging.defaultResponseHandler(eb, request, Contexts.CAMPAIGN.toString(), Actions.CREATE.toString(),
-                                null, campaign)));
+        RequestUtils.bodyToJson(request, pathPrefix + "campaign", campaign -> {
```
Comment (JinDouang) [inline src/main/java/fr/openent/crre/controllers/CampaignController.java:80]: lambda

File: src/main/java/fr/openent/crre/controllers/CampaignController.java [hunk 4]
```diff
@@ -74,10 +77,22 @@ public void campaign(HttpServerRequest request) {
...
+            UserUtils.getUserInfos(eb, request, userInfos -> {
+                try {
+                    campaignService.create(new Campaign(campaign), IModelHelper.toList(campaign.getJsonArray(Field.GROUPS), StructureGroupModel.class))
+                            .onSuccess(campaignResult -> {
+                                Renders.renderJson(request, campaignResult.toJson());
+                                Logging.insert(userInfos, Contexts.CAMPAIGN.toString(), Actions.CREATE.toString(), (String) null, campaign);
+                            })
+                            .onFailure(error -> Renders.renderError(request));
+                } catch (ClassCastException e) {
+                    log.error(String.format("[CRRE@%s::create] Fail to create campaign %s"),
```
Comment (JinDouang) [inline src/main/java/fr/openent/crre/controllers/CampaignController.java:90]: Attention ton string format est mal géré 

Aussi, tu ne devrais pas mettre `e.getMessage()` ?

File: src/main/java/fr/openent/crre/controllers/CampaignController.java [hunk 5]
```diff
@@ -107,17 +122,21 @@ public void updateAccessibility(final HttpServerRequest request) {
...
+                try {
+                    Integer id = Integer.parseInt(request.params().get(Field.ID));
+                    campaignService.update(new Campaign(campaign).setId(id), IModelHelper.toList(campaign.getJsonArray(Field.GROUPS), StructureGroupModel.class))
+                            .onSuccess(campaignResult -> {
+                                Renders.renderJson(request, campaignResult.toJson());
+                                Logging.insert(userInfos, Contexts.CAMPAIGN.toString(), Actions.UPDATE.toString(), id.toString(), campaign);
+                            })
+                            .onFailure(error -> Renders.renderError(request));
+                } catch (ClassCastException e) {
+                    log.error(String.format("[CRRE@%s::update] Fail to update campaign %s"),
```
Comment (JinDouang) [inline src/main/java/fr/openent/crre/controllers/CampaignController.java:135]: Attention ton string format est mal géré

Aussi, tu ne devrais pas mettre e.getMessage() ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**